File: backend/src/main/java/com/bomnuocv1/domain/repository/FarmerRepository.java
package com.bomnuocv1.domain.repository;

import com.bomnuocv1.domain.entity.Farmer;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface FarmerRepository {

    Farmer save(Farmer farmer);

    Optional<Farmer> findById(UUID id);

    List<Farmer> findByOwnerId(UUID ownerId);

    List<Farmer> searchByOwnerId(UUID ownerId, String keyword);

    boolean existsByOwnerIdAndPhoneNumber(UUID ownerId, String phoneNumber);

    boolean existsByOwnerIdAndPhoneNumberAndIdNot(UUID ownerId, String phoneNumber, UUID excludeId);

    long countByOwnerId(UUID ownerId);
}

File: frontend/app/src/main/java/com/vn/bomnuocv1/domain/repository/FarmerRepository.kt
package com.vn.bomnuocv1.domain.repository

import com.vn.bomnuocv1.domain.model.Farmer

interface FarmerRepository {

    suspend fun getFarmers(keyword: String? = null): Result<List<Farmer>>

    suspend fun createFarmer(fullName: String, phoneNumber: String?, areaNote: String?): Result<Farmer>

    suspend fun updateFarmer(id: String, fullName: String, phoneNumber: String?, areaNote: String?): Result<Farmer>

    suspend fun deleteFarmer(id: String): Result<Unit>
}
